include/plugin.hpp

```diff
@@ -123,18 +129,201 @@ class meminfo_plugin : public scorep::plugin::base<meminfo_plugin,
     template <typename Cursor>
     void get_all_values(const int& id, Cursor& c)
     {
-        for (const auto& data_point : data_) {
-            c.write(data_point.timepoint, data_point.data.find(id)->second.size);
+        auto handle = data_.find(id)->second;
+
+        for (int i = 0; i < handle.size(); ++i) {
+            c.write(times_[i], handle[i]);
         }
     }
 
 private:
     std::map<std::string, std::int64_t> subscribed_;
     std::atomic<bool> running = false;
     std::thread thread_;
-    std::vector<std::string> names_;
-    std::vector<dataset> data_;
+    std::map<std::int64_t, std::vector<std::int64_t>> data_;
+    std::vector<scorep::chrono::ticks> times_;
     std::chrono::nanoseconds intervall_;
     std::chrono::system_clock::time_point last_measurement_ =
         std::chrono::system_clock::now();
+
+    int mem_total_pos;
+    int mem_free_pos;
+    int buffers_pos;
+    int cache_pos;
+    int swap_total_pos;
+    int swap_free_pos;
+    int swap_cached_pos;
+    int mem_used_pos;
+    int swap_used_pos;
+
+    std::regex regex_parse =
+        std::regex(".*:[^a-zA-Z0-9]*([0-9]+).?([kKmMgGtT][bB])?.*");
+
+    std::map<std::int64_t, std::pair<std::string, std::string>> init(const std::vector<std::string>& search)
+    {
+        std::string line;
+        std::ifstream meminfo("/proc/meminfo");
+        std::map<std::int64_t, std::pair<std::string, std::string>> results;
+
+        std::string regex_custom_str;
+        std::string regex_str = "(";
+
+        if (search.empty()) {
+            regex_str += "[a-zA-Z0-9_]+";
+        }
+        else {
+            bool first = true;
+            for (const auto& s : search) {
+                if (first)
+                    first = false;
+                else {
+                    regex_str += '|';
+                }
+
+                regex_str += s;
+            }
+        }
+        regex_str += ")";
+        regex_custom_str = regex_str;
+        regex_str += ":[^a-zA-Z0-9]*([0-9]+).?([kKmMgGtT][bB])?[^a-zA-Z0-9]*";
+
+        std::regex regex(regex_str);
+
+        int mem_total;
+        int mem_free;
+        int buffers;
+        int cache;
+        int swap_total;
+        int swap_free;
+        int swap_cached;
+
+        std::int64_t line_nr = 0;
+        while (std::getline(meminfo, line)) {
+            std::smatch match;
+
+            bool save = false;
+
+            if (std::regex_match(line, match, regex)) {
+                save = true;
+            }
+            else if (std::regex_match(line, match, std::regex("(MemTotal|MemFree|SwapTotal|SwapFree|SwapCached|Cache|Buffers):[^a-zA-Z0-9]*([0-9]+).?([kKmMgGtT][bB])?[^a-zA-Z0-9]*"))) {
+                save = true;
+            }
+
+            if (save) {
+                results.emplace(line_nr, std::pair{match[1].str(), match[3].str()});
```

Comment:
You typically write `kB` as unit here, but normalize it to `B` elsewhere. Should also normalize `kB` to `B` here.

Reply:
fixed